packages/circuit-ui/components/RadioButtonGroup/RadioButtonGroup.mdx

```diff
@@ -18,20 +18,102 @@ Use a radio button group when the user has to choose only one option based on a
 
 ## Usage guidelines
 
-- **Do** always associate the radio button with a label that clearly gives context for the user to make a decision
 - **Do** capitalize each item of the list
 - **Do** list the items in a rational order
 - **Do not** punctuate the end of every list item
 - **Do not** mark required fields with asterisks
 
-## Component variations
+## Validations
 
-### Validation States
+Use the `validationHint` to communicate the expected response to users. Use the `invalid` and `hasWarning` props to require or suggest a change. The `showValid` prop can be used to provide an extra confirmation that the response is valid.
 
-<Story of={Stories.Validation} />
+<Story of={Stories.Validations} />
 
-### Disabled radio button
+## Disabled
 
-Use a disabled radio button only if you really need to communicate to the user that an option that existed before is not available for choosing now. Keep in mind that you need to communicate why. One recommendation would be to pair a disabled input with a tooltip to give the appropriate context to the user.
+Disabled form fields can be confusing to users, so use them with caution. Use a disabled radio button only if you need to communicate to the user that an option that existed before is not available for choosing now. Consider not displaying the field at all or add an explanation why the field is disabled.
+
+The radio button group can be disabled entirely or in part.
 
 <Story of={Stories.Disabled} />
+
+## State
+
+The RadioButtonGroup can be used as a controlled or uncontrolled component.
+
+### Controlled
+
+```tsx
+import { useState, ChangeEvent } from 'react';
+import { RadioButtonGroup } from '@sumup/circuit-ui';
+
+function Controlled() {
+  const [value, setValue] = useState([]);
+
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setValue(event.target.value);
+  };
+
+  const options = [
+    // ...
+  ];
+
+  return (
+    <RadioButtonGroup
+      label="Favourite fruit"
+      options={options}
+      value={value}
+      onChange={onChange}
+    />
+  );
+}
+```
+
+### Uncontrolled
+
+```tsx
+import { useRef } from 'react';
+import { RadioButtonGroup } from '@sumup/circuit-ui';
+
+function Uncontrolled() {
+  const checkboxRefs = useRef({});
+
+  const setCheckboxRef = (input: HTMLInputElement) => {
+    checkboxRefs.current[input.value] = input;
+  };
+
+  const options = [
+    {
+      label: 'Apple',
+      value: 'apple',
+      ref: setCheckboxRef,
+    },
+  ];
+
+  return (
+    <RadioButtonGroup
+      label="Favourite fruit"
+      options={options}
+      defaultValue={[]}
+      ref={checkboxGroupRef}
+    />
+  );
+}
+```
+
+## Accessibility
+
+### Best practices
+
+#### Write clear and concise labels
+
+The `label` prop passed to the RadioButtonGroup or one of its `options` constitutes the input's label. This label should be clear and concise, and should not contain any formatted or semantic content (such as headings or lists).
```

Comment:
Nit: it's the group label for the `RadioButtonGroup` and the option's label for an `option`. Can be combined when announced. Both are important 🙂 (not sure whether we can make this clearer somehow)

Reply:
Clarified in 29468c6.